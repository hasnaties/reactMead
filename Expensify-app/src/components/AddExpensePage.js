import React from 'react';
import ExpenseForm from './ExpenseForm';

const AddExpensePage = () => (
  <div>
    <h4>Add Expense</h4>
    <ExpenseForm />
  </div>
);

export default AddExpensePage;