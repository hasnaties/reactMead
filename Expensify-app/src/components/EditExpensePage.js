import React from 'react';

const EditExpensePage = () => (
  <div>
    <p>Edit</p>
  </div>
);

export default EditExpensePage;