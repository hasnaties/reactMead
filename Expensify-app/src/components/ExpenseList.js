import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import ExpenseListItem from './ExpenseListItem';

export const ExpenseList = (props) => (
  
  <div>
    {props.expenses.length === 0 ? (
      <p>Expenses list is empty.</p>
    ) : (
      <div>
        <h4>Expenses List</h4>
        {
          props.expenses.map((expense) => {
          return (
            <ExpenseListItem key={expense.id} {...expense}/>
          );
          })
        }
      </div>
    )}
  </div>
);

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  }
}

export default connect(mapStateToProps)(ExpenseList)