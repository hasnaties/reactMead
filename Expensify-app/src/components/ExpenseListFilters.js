import React from 'react';
import { connect } from 'react-redux';
import { editTextFilter, setEndDate, setStartDate, sortByAmount, sortByDate } from '../actions/filters';
import { DateRangePicker } from 'react-dates';

export class ExpenseListFilters extends React.Component {
  
  state = {
    calendarFocused: null
  }

  onDatesChange = ({startDate, endDate}) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  }

  onFocusChange = (calendarFocused) => {
    this.setState({ calendarFocused })
  }

  onTextChange = (e) => {
    return (this.props.editTextFilter(e.target.value));
  }

  onSortByChange = (e) => {
    if(e.target.value === "date") {
      return (this.props.sortByDate());
    }
    return (this.props.sortByAmount());
  }
  
  render () {
    return (
      <div>
        <input type='text' 
        value={this.props.filters.text} 
        onChange={this.onTextChange}>
        </input>
    
        <select 
        value={this.props.filters.sortBy} 
        onChange={this.onSortByChange}
        >
          <option value="date">Date</option>
          <option value="amount">Amount</option>
        </select>
        
        <DateRangePicker
          startDate={this.props.filters.startDate}
          endDate={this.props.filters.endDate}
          onDatesChange={this.onDatesChange}
          focusedInput={this.state.calendarFocused}
          onFocusChange={this.onFocusChange}
          numberOfMonths={1}
          isOutsideRange={() => false}
          showClearDates={true}
        />
      </div>
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
  
  setStartDate: (startDate) => dispatch(setStartDate(startDate)),
  setEndDate: (endDate) => dispatch(setEndDate(endDate)),
  editTextFilter: (value) => dispatch(editTextFilter(value)),
  sortByDate: () => dispatch(sortByDate()),
  sortByAmount: () => dispatch(sortByAmount())
});

const mapStateToProps = (props) => ({
  filters: props.filters
});

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters)