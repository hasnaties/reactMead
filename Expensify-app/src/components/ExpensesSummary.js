import React from 'react';
import { connect } from "react-redux";
import selectExpenses from '../selectors/expenses';
import expensesTotal from '../selectors/expenses-total';
import numeral from "numeral";

export const ExpensesSummary = (props) => (
  <div>
    <p>
      Viewing {props.expenseCount} expense{props.expenseCount > 1 ? 's' : undefined} totalling {numeral(props.expensesTotal / 100).format('$0,0.00')}
    </p>
  </div>
);

const mapStateToProps = (state) => ({
  expenseCount: selectExpenses(state.expenses, state.filters).length,
  expensesTotal: expensesTotal(selectExpenses(state.expenses, state.filters))
});

export default connect(mapStateToProps)(ExpensesSummary);