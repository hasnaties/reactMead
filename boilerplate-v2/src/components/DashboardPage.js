import React from 'react';

const ExpenseDashboardPage = () => (
  <div>
    Dashboard content.
  </div>
);

export default ExpenseDashboardPage;