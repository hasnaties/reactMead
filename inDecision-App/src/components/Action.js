import React from "react";

const Action = (props) => (
    <div>
        <button 
            className="big-button" 
            onClick= {props.handleAction} 
            disabled= {!(props.btnState)}
            >
            What should I do?
        </button>
    </div>
);

export default Action;