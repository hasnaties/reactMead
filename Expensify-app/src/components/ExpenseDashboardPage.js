import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpensesSummary from './ExpensesSummary';

const ExpenseDashboardPage = () => (
  <div>
    <p>Dashboard.</p>
    <ExpensesSummary />
    <ExpenseListFilters />
    <ExpenseList />
  </div>
);

export default ExpenseDashboardPage;