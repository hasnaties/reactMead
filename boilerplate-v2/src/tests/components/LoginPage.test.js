import React from "react";
import { shallow } from "enzyme";
import { LoginPage } from "../../components/LoginPage";

let startLogin = jest.fn();
let wrapper = shallow(<LoginPage startLogin={startLogin}/>);

test('should Render the LoginPage correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
test('should Handle login button', () => {

  wrapper.find('button').simulate('click');
  expect(startLogin).toHaveBeenCalled();
});