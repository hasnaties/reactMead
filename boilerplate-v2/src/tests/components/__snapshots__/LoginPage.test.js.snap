// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should Render the LoginPage correctly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Boilerplate React v2
    </h1>
    <p>
      This is the sub-title.
    </p>
    <button
      className="button"
      onClick={[Function]}
    >
      Login with Google
    </button>
  </div>
</div>
`;
